<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="input" default="/camera/color/image_raw"/>
    <arg name="output" default="/yolo/pred"/>
    <arg name="weights" default="$(find open_vocab_vision_ros)/checkpoints/yolov8x-worldv2.pt"/>
    <arg name="drop" default="true"/>
    <arg name="confidence" default="0.3"/>
    <arg name="labels" 
	    default="car,person,door,chair,table,robot,street sign,sign,stop sign,lamp pole,trash can,television,monitor,desk,traffic cone,table,box,package,machine,plant"/>

    <arg name="viz" default="true"/>
    
    <!-- don't consider detections farther than this -->
    <arg name="depth_threshold" default="7.5"/>

    <!-- cluster detections within this threshold into same object-->
    <arg name="distance_threshold" default="1"/>
    <!-- sleep call between detections -->

    <!-- sleep call between detections -->
    <arg name="detect_period" default="0.25"/>

    <arg name="n_track_thresh" default="5"/>
 
    <include file="$(find open_vocab_vision_ros)/launch/yolo.launch" pass_all_args="true"/>

</launch>
